### PR TITLE
Drop prospector; use pylint directly

### PR DESCRIPTION
**Pull request by berkerpeksag**

### CODE REVIEW

**Comment by tilgovi** [comment]
Have to figure out why the build is failing, but the actual changes here seem fine.

**Comment by berkerpeksag** [comment]
https://github.com/benoitc/gunicorn/pull/1678/commits/9a4f9974d748f94e59e24dfe2ebc5794f1284545 broke the build. I will try to find a way to check tests/ directory without adding `tests/__init__.py`.

**Comment by tilgovi** [comment]
@berkerpeksag I am dealing with the same issue with another project right now. `pylint` expects individual modules or a package, so to get it to check `tests` we have to pass them all as file paths rather than passing the `tests` directory, unfortunately.

**Comment by berkerpeksag** [comment]
@tilgovi I tried to pass `tests/*` but that didn't work because it ignored the `ignore` setting (test_selectors.py etc.) in `.pylintrc`. I will try to pass all of them manually. I hate all linters :D

**Comment by tilgovi** [comment]
@berkerpeksag I am also fine, if it unblocks this, to avoid linting tests for now.

**Comment by berkerpeksag** [comment]
Open issues:

```
************* Module gunicorn.errors
W: 10, 4: __init__ method from base class 'BaseException' is not called (super-init-not-called)
************* Module gunicorn.app.base
W: 79, 0: Method 'init' is abstract in class 'BaseApplication' but is not overridden (abstract-method)
W: 79, 0: Method 'load' is abstract in class 'BaseApplication' but is not overridden (abstract-method)
************* Module gunicorn.app.pasterapp
W: 71, 0: Method 'init' is abstract in class 'BaseApplication' but is not overridden (abstract-method)
W: 71, 0: Method 'load' is abstract in class 'BaseApplication' but is not overridden (abstract-method)
W:121, 0: Method 'init' is abstract in class 'BaseApplication' but is not overridden (abstract-method)
W:123, 4: Keyword argument before variable positional arguments list in the definition of __init__ function (keyword-arg-before-vararg)
W:123, 4: __init__ method from base class 'PasterBaseApplication' is not called (super-init-not-called)
W:185, 0: Keyword argument before variable positional arguments list in the definition of paste_server function (keyword-arg-before-vararg)
************* Module gunicorn.http.errors
W: 12, 4: __init__ method from base class 'OSError' is not called (super-init-not-called)
W: 20, 4: __init__ method from base class 'ParseException' is not called (super-init-not-called)
W: 29, 4: __init__ method from base class 'ParseException' is not called (super-init-not-called)
W: 37, 4: __init__ method from base class 'ParseException' is not called (super-init-not-called)
W: 45, 4: __init__ method from base class 'ParseException' is not called (super-init-not-called)
W: 54, 4: __init__ method from base class 'ParseException' is not called (super-init-not-called)
W: 62, 4: __init__ method from base class 'OSError' is not called (super-init-not-called)
W: 70, 4: __init__ method from base class 'OSError' is not called (super-init-not-called)
W: 78, 4: __init__ method from base class 'ParseException' is not called (super-init-not-called)
W: 87, 4: __init__ method from base class 'ParseException' is not called (super-init-not-called)
W: 95, 4: __init__ method from base class 'ParseException' is not called (super-init-not-called)
W:104, 4: __init__ method from base class 'ParseException' is not called (super-init-not-called)
************* Module gunicorn.http.message
W: 24, 0: Anomalous backslash in string: '\['. String constant might be missing an r prefix. (anomalous-backslash-in-string)
W: 24, 0: Anomalous backslash in string: '\]'. String constant might be missing an r prefix. (anomalous-backslash-in-string)
************* Module gunicorn.http.wsgi
W: 54, 4: __init__ method from base class 'RawIOBase' is not called (super-init-not-called)
************* Module gunicorn.instrument.statsd
W: 30, 0: Anomalous backslash in string: '\g'. String constant might be missing an r prefix. (anomalous-backslash-in-string)
```

File: tox.ini [hunk 1]
```diff
@@ -12,6 +12,21 @@ deps =
...
+    tests/test_http.py \
+    tests/test_invalid_requests.py \
+    tests/test_logger.py \
+    tests/test_pidfile.py \
+    tests/test_sock.py \
+    tests/test_ssl.py \
+    tests/test_statsd.py \
+    tests/test_systemd.py \
+    tests/test_util.py \
+    tests/test_valid_requests.py
```
Comment (tilgovi) [inline tox.ini]: 😆 😭

**Comment by berkerpeksag** [comment]
I think we can ignore all super-init-not-called warnings. Normally `BaseException.__str__` uses the first element of `*args`. However, we also define our own `__str__` in all `Exception` subclasses so we don't need to `super()` in `__init__` methods.

**Comment by berkerpeksag** [comment]
Remaining ones:

```
************* Module gunicorn.app.base
W: 79, 0: Method 'init' is abstract in class 'BaseApplication' but is not overridden (abstract-method)
W: 79, 0: Method 'load' is abstract in class 'BaseApplication' but is not overridden (abstract-method)
************* Module gunicorn.app.pasterapp
W: 71, 0: Method 'init' is abstract in class 'BaseApplication' but is not overridden (abstract-method)
W: 71, 0: Method 'load' is abstract in class 'BaseApplication' but is not overridden (abstract-method)
W:121, 0: Method 'init' is abstract in class 'BaseApplication' but is not overridden (abstract-method)
W:123, 4: Keyword argument before variable positional arguments list in the definition of __init__ function (keyword-arg-before-vararg)
W:123, 4: __init__ method from base class 'PasterBaseApplication' is not called (super-init-not-called)
W:185, 0: Keyword argument before variable positional arguments list in the definition of paste_server function (keyword-arg-before-vararg)
************* Module gunicorn.http.message
W: 24, 0: Anomalous backslash in string: '\['. String constant might be missing an r prefix. (anomalous-backslash-in-string)
W: 24, 0: Anomalous backslash in string: '\]'. String constant might be missing an r prefix. (anomalous-backslash-in-string)
************* Module gunicorn.http.wsgi
W: 54, 4: __init__ method from base class 'RawIOBase' is not called (super-init-not-called)
************* Module gunicorn.instrument.statsd
W: 30, 0: Anomalous backslash in string: '\g'. String constant might be missing an r prefix. (anomalous-backslash-in-string)
```

@benoitc, @tilgovi: I'm not good with regex so I need help with anomalous-backslash-in-string warnings.

**Comment by berkerpeksag** [comment]
@tilgovi I need your comment on https://github.com/benoitc/gunicorn/pull/1678/commits/4114b0212f8a1772ecda9699c6835de85a022fd4 too.

**Comment by berkerpeksag** [comment]
```
Using config file /home/berker/hacking/gunicorn/.pylintrc
************* Module gunicorn.app.base
W: 79, 0: Method 'init' is abstract in class 'BaseApplication' but is not overridden (abstract-method)
W: 79, 0: Method 'load' is abstract in class 'BaseApplication' but is not overridden (abstract-method)
************* Module gunicorn.app.pasterapp
W: 71, 0: Method 'init' is abstract in class 'BaseApplication' but is not overridden (abstract-method)
W: 71, 0: Method 'load' is abstract in class 'BaseApplication' but is not overridden (abstract-method)
W:121, 0: Method 'init' is abstract in class 'BaseApplication' but is not overridden (abstract-method)
************* Module gunicorn.http.message
W: 24, 0: Anomalous backslash in string: '\['. String constant might be missing an r prefix. (anomalous-backslash-in-string)
W: 24, 0: Anomalous backslash in string: '\]'. String constant might be missing an r prefix. (anomalous-backslash-in-string)
************* Module gunicorn.http.wsgi
W: 54, 4: __init__ method from base class 'RawIOBase' is not called (super-init-not-called)
************* Module gunicorn.instrument.statsd
W: 30, 0: Anomalous backslash in string: '\g'. String constant might be missing an r prefix. (anomalous-backslash-in-string)
```

**Comment by tilgovi** [comment]
```diff
diff --git a/gunicorn/http/message.py b/gunicorn/http/message.py
index acd7387..3b8e77e 100644
--- a/gunicorn/http/message.py
+++ b/gunicorn/http/message.py
@@ -21,7 +21,7 @@ MAX_REQUEST_LINE = 8190
 MAX_HEADERS = 32768
 DEFAULT_MAX_HEADERFIELD_SIZE = 8190
 
-HEADER_RE = re.compile("[\x00-\x1F\x7F()<>@,;:\[\]={} \t\\\\\"]")
+HEADER_RE = re.compile(r"[\x00-\x1F\x7F()<>@,;:\[\]={} \t\\\"]")
 METH_RE = re.compile(r"[A-Z0-9$-_.]{3,20}")
 VERSION_RE = re.compile(r"HTTP/(\d+)\.(\d+)")
 
diff --git a/gunicorn/instrument/statsd.py b/gunicorn/instrument/statsd.py
index e0c6882..4bbcb20 100644
--- a/gunicorn/instrument/statsd.py
+++ b/gunicorn/instrument/statsd.py
@@ -27,7 +27,7 @@ class Statsd(Logger):
         """host, port: statsD server
         """
         Logger.__init__(self, cfg)
-        self.prefix = sub(r"^(.+[^.]+)\.*$", "\g<1>.", cfg.statsd_prefix)
+        self.prefix = sub(r"^(.+[^.]+)\.*$", "\\g<1>.", cfg.statsd_prefix)
         try:
             host, port = cfg.statsd_host
             self.sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
```

**Comment by tilgovi** [comment]
I would just `# pylint: ignore` the errors about the Paster abstract methods. We'll likely be removing that code in R20.

**Comment by berkerpeksag** [comment]
@tilgovi I've addressed all of your comments and we finally have a green build!

**Pull request merged**